### ISSUE TITLE
how to load basic programs on the chip?

### ISSUE THREAD

**Issue by pabloxid** [aanon4/BlueBasic#30]
Is this "console" app. mean to load basic programs on the chip? Is there a Windows version?

**Comment by 0xFACE** [comment]
Yes, with console app you can load, list, edit programs on the chip. Unfortunately there is no Windows version.

**Comment by 0x2b3bfa0** [comment]
@pabloxid: You can try [BlueBasic-loader](https://github.com/0xFACE/BlueBasic-loader). However it's for Linux, not Windows. Maybe you're interested in installing a Linux distribution like Ubuntu... I use it as my main operative system and I don't miss any functionality form M$ Windows 😈

**Comment by ozarchie** [comment]
I think you can also use Botspine to do it. Although it complains that the
devices are not botspines, you can still load basic programs.

**Comment by pabloxid** [comment]
Thanks for your answers.
I have a couple more questions:
1) how the console actually communicates with the device? Via Bluetooth?
2) Can I upload the firmware to the device with a basic program already included, using CC254x debugger?
3) there is some documentation on the console-device protocol?
Thank you again.

**Comment by 0x2b3bfa0** [comment]
@pabloxid:
1. Yes, vía Bluetooth 4.0
2. I don't know, however somebody here could answer that.
3. Take a look to the script linked in my last message. You can see how it works. However, You could ask the autor for a detailed description.

**Comment by ozarchie** [comment]
Unfortunately, OSX-Sierra completely breaks the console. Don't upgrade if you don't need to.

**Comment by kscheff** [comment]
For a working OSX Sierra version see my fork starting at #31

**Comment by TheSmartGerman** [comment]
I'm currently searing for a app to download Basic Programms (Android / Windows would be fine).

I found this: https://play.google.com/store/apps/details?id=de.kai_morich.serial_bluetooth_terminal

you can configure the UUID Service, Sending and Receiving UUID.

The Console App out of the repository does the same:

```
140              if list[UUIDS.commsServiceUUID] != nil { 
141                self.inputCharacteristic = list[UUIDS.commsServiceUUID]!.characteristics[UUIDS.inputCharacteristicUUID] 
142                self.outputCharacteristic = list[UUIDS.commsServiceUUID]!.characteristics[UUIDS.outputCharacteristicUUID] 
```

**Comment by TheSmartGerman** [comment]
You can use the Android Bluetooth Serial Terminal, it allows to use custom UUIDs: 
https://play.google.com/store/apps/details?id=de.kai_morich.serial_bluetooth_terminal&hl=de

It's working with the hex from this fork!

**Comment by ozarchie** [comment]
I did this a while ago, but gave up on the CC2540 in favour of ESP8266
https://github.com/ozarchie/BlueBasic-loader

**Comment by TheSmartGerman** [comment]
Well, I found a android solution. It's fine for me. 

I just mentioned it here. But beware using the branch from kscheff, does have a problem with the cc2541 build: https://github.com/kscheff/BlueBasic/issues/8

The ESP8266 / ES32 a known but right know no option....
